src/main/kotlin/net/atos/zac/app/zaken/ZakenRESTService.kt

```diff
@@ -341,13 +351,13 @@ class ZakenRESTService {
                 zaak
             )
         }
-        restZaak.groep?.let { restGroup ->
-            val group = identityService.readGroup(restGroup.id)
-            zrcClientService.updateRol(zaak, bepaalRolGroep(group, zaak), AANMAKEN_ZAAK_REDEN)
+        restZaak.groep?.let {
```

Comment:
I know.. this is just the beginning... many places to improve the code for sure. I suggest to do that in other PRs.

Reply:
> This can also be written like so, a bit more functional ...
> 
> ```
> restZaak.groep?.id?
>     .let { identityService.readGroup(it) }
>     .let { zakenService.bepaalRolGroep(it, zaak) }
>     .let { zrcClientService.updateRol(zaak, it, AANMAKEN_ZAAK_REDEN) }
> ```

Good idea. But this breaks our current unit tests so I will look at this in another PR if you don't mind. :-)